rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -136,13 +136,28 @@ impl HashAggregateExec {
     /// Get new orphan of execution plan
     pub fn new_orphan(&self) -> Arc<HashAggregateExec> {
         Arc::new(HashAggregateExec {
-            input: Arc::new(DummyExec {}), 
+            input: Arc::new(DummyExec {}),
             mode: self.mode.clone(),
             group_expr: self.group_expr.clone(),
             aggr_expr: self.aggr_expr.clone(),
             schema: self.schema.clone(),
         })
     }
+
+    /// Get the group_expr field of this plan
+    pub fn group_expr(&self) -> Vec<(Arc<dyn PhysicalExpr>, String)> {
```

Comment:
shall we add a new function like this:

```rust
pub fn new(&self, plan: Arc<dyn ExecutionPlan>) -> Arc<HashAggregateExec> {
        Arc::new(HashAggregateExec {
            input: plan.clone(), 
            mode: self.mode.clone(),
            group_expr: self.group_expr.clone(),
            aggr_expr: self.aggr_expr.clone(),
            schema: self.schema.clone(),
        })
    }
```

Reply:
Yeah. I think it'll be better that way. 1 `.clone()` is better than 2 `.clone()`s. I'll commit a new version.